pygmt/helpers/decorators.py

```diff
@@ -575,6 +575,27 @@ def new_module(*args, **kwargs):
                         f"Use long-form parameter '{long_alias}' instead."
                     )
                     warnings.warn(msg, category=SyntaxWarning, stacklevel=2)
+
+            # xshift (X) is deprecated since v0.8.0.
+            if "X" in kwargs or "xshift" in kwargs:
+                if "xshift" in kwargs:
+                    kwargs["X"] = kwargs.pop("xshift")
+                msg = (
+                    "Parameters 'X' and 'xshift' are deprecated since v0.8.0. "
+                    "Use Figure.shift_origin(xshift=...) instead."
+                )
```

Comment:
Do we want to fully remove these parameters in future versions (e.g., v0.12.0)?
```suggestion
                msg = (
                    "Parameters 'X' and 'xshift' are deprecated since v0.8.0 "
                    "and will be removed in v0.12.0. "
                    "Use Figure.shift_origin(xshift=...) instead."
                )
```

Reply:
4 minor versions sounds good.